test/e2e/handler/broken_configuration_test.go

```diff
@@ -0,0 +1,171 @@
+			if err != nil {
+				return
+			}
+
+			By("Waiting until the node becomes ready again")
+			waitForNodesReady()
+
+			By("Waiting for policy to be ready")
+			waitForDegradedPolicy(DefaultNetworkBroken)
+
+		})
+		AfterEach(func() {
+			By("Remove the policy")
+			deletePolicy(DefaultNetworkBroken)
+
+			By("Reset desired state at all nodes")
+			resetDesiredStateForNodes()
+		})
+		It("should have the policy with fail message", func() {
+
+			for _, failMessage := range failMessages {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+
+			for _, failMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(failMessage))
+			}
+		})
+	})
+
+	Context("with invalid interface name", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateLibnmError:",
+				"No suitable device found for this connection",
+			}
+		)
+		BeforeEach(func() {
+			By("Creating the policy")
+			err := setDesiredStateWithPolicyAndNodeSelector(DefaultNetworkBroken, createInterfaceWithInvalidName(), map[string]string{"kubernetes.io/hostname": "node01"})
+			if err != nil {
+				return
+			}
+
+			By("Waiting until the node becomes ready again")
+			waitForNodesReady()
+
+			By("Waiting for policy to be ready")
+			waitForDegradedPolicy(DefaultNetworkBroken)
+
+		})
+		AfterEach(func() {
+			By("Remove the policy")
+			deletePolicy(DefaultNetworkBroken)
+
+			By("Reset desired state at all nodes")
+			resetDesiredStateForNodes()
+		})
+		It("should have the policy with fail message", func() {
+
+			for _, failMessage := range failMessages {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+
+			for _, validMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(validMessage))
+			}
+		})
+	})
+
+	Context("with ping fail", func() {
+		var (
+			messagesToKeep = [...]string{
+				"failed to retrieve default gw at runProbes",
+			}
+		)
+		BeforeEach(func() {
+			By("Creating the policy")
+			err := setDesiredStateWithPolicyAndNodeSelector(DefaultNetworkBroken, createInterfaceWithPingFail(), map[string]string{"kubernetes.io/hostname": "node01"})
+			if err != nil {
+				return
+			}
+
+			By("Waiting until the node becomes ready again")
+			waitForNodesReady()
+
+			By("Waiting for policy to be ready")
+			waitForDegradedPolicy(DefaultNetworkBroken)
+
+		})
+
+		AfterEach(func() {
```

Comment:
If I'm not mistaken, the `AfterEach` is always the same, you may just define it one level higher (in the Describe block), so that it's applied for all `Context`s

Reply:
I removed this part completely as it caused problems